### PR TITLE
test(docs,website): add automated lighthouse audits

### PR DESCRIPTION
**Pull request by MFarabi619**
## PR Requirements Checklist

- [x] Commit messages follow [guidelines](https://docs.cuhacking.ca/contribution-guidelines/conventional-commits)
- [x] Tests added
- [x] Tests pass
- [ ] Docs added/updated (for bug fixes/features)
- [x] Rebased onto main
- [x] Tested for mobile/tablet/desktop (if applicable)

### Steps
- Run `pnpm nx e2e docs-e2e --ui` 
- Run the lighthouse test and let it complete
- `pnpm lighthouse` to see the report :)

Lighthouse test for `pnpm nx e2e website-e2e --ui` has been disabled for now as the Spline component hangs up the main thread and fails the test consistently.

## Release Notes

- **New Features**
  - Introduced parallel test execution for Playwright tests, improving efficiency.
  - Added Lighthouse audits integrated into end-to-end tests, enhancing performance evaluation.
  - New script for generating and viewing Lighthouse reports.

- **Bug Fixes**
  - Updated web server command for local development.

- **Chores**
  - Added entries to `.gitignore` for `playwright-report` and `lighthouse-report`.
  - Removed outdated test file `example.spec.ts`.
  - Added new dependency `playwright-lighthouse` for improved testing capabilities.
  - Updated end-to-end test workflow configuration to reflect new test types.
  - Updated project configurations to manage dependencies more effectively.
  - Added implicit dependency for the `docs-e2e` project.
  - Commented out tests for links requiring authentication in `docs.spec.ts`.

### CODE REVIEW

File: apps/docs-e2e/src/lighthouse.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    port,
+    thresholds: thresholdsConfig,
+    reports: {
+      formats: {
+        // json: true, // defaults to false
+        html: true, // defaults to false
+        // csv: true, // defaults to false
+      },
+      name: `latest-report`, // defaults to `lighthouse-${new Date().getTime()}`
+      directory: `${process.cwd()}../../../lighthouse-report`, // defaults to `${process.cwd()}/lighthouse`
```
[the remark this answers is not in the document]
Comment (MFarabi619) [reply apps/docs-e2e/src/lighthouse.spec.ts:50]: This is fine, I had issues after adding the slash.

File: apps/website-e2e/src/website.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    port,
+    thresholds: thresholdsConfig,
+    reports: {
+      formats: {
+        // json: true, // defaults to false
+        html: true, // defaults to false
+        // csv: true, // defaults to false
+      },
+      name: `latest-report`, // defaults to `lighthouse-${new Date().getTime()}`
+      directory: `${process.cwd()}../../../lighthouse-report`, // defaults to `${process.cwd()}/lighthouse`
```
[the remark this answers is not in the document]
Comment (MFarabi619) [reply apps/website-e2e/src/website.spec.ts:50]: This is fine, I had issues after adding the slash.

**Comment by JeremyFriesenGitHub** [comment]
🔥

**Pull request merged**